using-mobile-extensions/adobe-experience-platform-messaging/README.md

```diff
@@ -0,0 +1,211 @@
+
+{% endtab %}
+{% endtabs %}
+
+## Add the AEP Messaging extension to your application
+
+### Download and import the Messaging extension
+
+{% tabs %}
+{% tab title="Android" %}
+
+### Java
+
+1. Add the Mobile Core, Edge and Messaging extensions to your project using the app's Gradle file.
+
+   ```java
+   implementation 'com.adobe.marketing.mobile:core:1.+'
+   implementation 'com.adobe.marketing.mobile:edge:1.+'
+   implementation 'com.adobe.marketing.mobile:edgeidentity:1.+'
+   implementation 'com.adobe.marketing.mobile:messaging:1.+'
+   ```
+
+2. Import the Mobile Core Edge and Messaging extensions in your application class.
+
+   ```java
+    import com.adobe.marketing.mobile.*;
+   ```
+
+
+{% endtab %}
+
+{% tab title="iOS" %}
+
+1. Add the Mobile Core, Edge and Messaging extensions to your project using Cocoapods. Add following pods in your `Podfile`:
+
+   ```swift
+   use_frameworks!
+   target 'YourTargetApp' do
+       pod 'AEPCore'
+       pod 'AEPEdge'
+       pod 'AEPEdgeIdentity'
+       pod 'AEPMessaging'
+   end
+   ```
+
+2. Import the Mobile Core, Edge and Messaging libraries:
+
+### Swift
+
+```swift
+// AppDelegate.swift
+import AEPCore
+import AEPEdge
+import AEPEdgeIdentity
+import AEPMessaging
+```
+
+### Objective-C
+
+```text
+// AppDelegate.h
+@import AEPCore;
+@import AEPEdge;
+@import AEPEdgeIdentity;
+@import AEPMessaging;
+```
+
+{% endtab %}
+{% endtabs %}
+
+### Register Messaging extensions with Mobile Core
+
+{% tabs %}
+{% tab title="Android" %}
+
+### Java
+
+```java
+public class MobileApp extends Application {
+    @Override
+    public void onCreate() {
+      super.onCreate();
+      MobileCore.setApplication(this);
+      MobileCore.configureWithAppID("yourLaunchEnvironmentID");
+      try {
+        Edge.registerExtension();
+        Identity.registerExtension();
+        AEPMessaging.registerExtension(); // register Messaging
+        MobileCore.start(new AdobeCallback() {
+          @Override
+          public void call(final Object o) {
+            // processing after start
+          }});
+      } catch (Exception e) {
+        //Log the exception
+      }
+    }
+}
+```
+
+{% endtab %}
+
+{% tab title="iOS" %}
+
+### Swift
+
+```swift
+// AppDelegate.swift
+func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+    MobileCore.registerExtensions([Identity.self, Edge.self, AEPMessaging.self], {
+    	MobileCore.configureWith(appId: "yourLaunchEnvironmentID")
+    })
+  ...
+}
+```
+
+### Objective-C
+
+```objective-c
+// AppDelegate.m
+- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+    [AEPMobileCore registerExtensions:@[AEPMobileEdgeIdentity.class, AEPMobileEdge.class, AEPMobileMessaging.class] completion:^{
+    [AEPMobileCore configureWithAppId: @"yourLaunchEnvironmentID"];
+  }];
+  ...
+}
+```
+
+{% endtab %}
+{% endtabs %}
+
```

Comment:
Do we also have to call setPushIdentifier as a step while setting up? Is that necessary?

Reply:
I have added that in the API i am not sure if we want to add in setup or in api? @sbenedicadb any suggestion

Reply:
Wondering if customers would miss this step and stop after registration. If we are planning on having a step-by-step guide it would be fine I guess